src/v/debug_bundle/debug_bundle_service.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "config/property.h"
+#include "debug_bundle/error.h"
+
+#include <seastar/core/sharded.hh>
+
+namespace debug_bundle {
+
+/**
+ * @brief Service used to manage creation of debug bundles
+ *
+ * This service is used to create, delete, and manage debug bundles using the
+ * "rpk debug bundle" application
+ */
+class debug_bundle_service final
+  : public ss::peering_sharded_service<debug_bundle_service> {
```

Comment:
It looks like this should only run on a single core, is peering or sharded necessary?

Reply:
> I was trying to be helpful to whichever other service will use this service (namely admin api). It ensures that the using service doesn't have to know which shard this service operates on.

Makes sense, there's not currently too much state that's redundant on other cores.